addons/im_livechat/static/src/models/public_livechat_window.js

```diff
@@ -1,14 +1,41 @@
 /** @odoo-module **/
 
+import PublicLivechatWindow from '@im_livechat/legacy/widgets/public_livechat_window/public_livechat_window';
+
 import { registerModel } from '@mail/model/model_core';
 import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
 
 registerModel({
     name: 'PublicLivechatWindow',
     identifyingFields: ['livechatButtonViewOwner'],
     lifecycleHooks: {
         _willDelete() {
-            this.legacyChatWindow.destroy();
+            if (this.legacyChatWindow) {
+                this.legacyChatWindow.destroy();
+            }
+        },
+    },
+    recordMethods: {
+        /**
+         * @private
+         * @returns {im_livechat/legacy/widgets/public_chat_window|FieldCommand}
+         */
+        _computeLegacyChatWindow() {
+            if (!this.livechatButtonViewOwner.widget && this.legacyChatWindow) {
+                this.legacyChatWindow.destroy();
```

Comment:
side effect inside a compute?

Reply:
Yes, the model itself is responsible of destroying the legacy window. Can't do much better here because this is all legacy code.